google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddCombinedRuleUserList.java

```diff
@@ -0,0 +1,175 @@
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    String urlString = "url__";
+
+    // Creates a rule targeting any user that visited a url that equals 'example.com/example1'.
+    UserListRuleItemInfo rule1 =
+        UserListRuleItemInfo.newBuilder()
+            // Uses a built-in parameter to create a domain URL rule.
+            .setName(StringValue.of(urlString))
+            .setStringRuleItem(
+                UserListStringRuleItemInfo.newBuilder()
+                    .setOperator(UserListStringRuleItemOperator.EQUALS)
+                    .setValue(StringValue.of("example.com/example1"))
+                    .build())
+            .build();
+
+    // Creates a rule targeting any user that visited a url that equals 'example.com/example2'.
+    UserListRuleItemInfo rule2 =
+        UserListRuleItemInfo.newBuilder()
+            // Uses a built-in parameter to create a domain URL rule.
+            .setName(StringValue.of(urlString))
+            .setStringRuleItem(
+                UserListStringRuleItemInfo.newBuilder()
+                    .setOperator(UserListStringRuleItemOperator.EQUALS)
+                    .setValue(StringValue.of("example.com/example2"))
+                    .build())
+            .build();
+
+    // Creates an ExpressionRuleUserListInfo object specifying that both rule1 and rule2 must be
+    // satisfied for a user to be added to this list.
+    ExpressionRuleUserListInfo expressionRuleUserListInfo =
+        ExpressionRuleUserListInfo.newBuilder()
+            .setRule(
+                // The default rule_type for a UserListRuleInfo object is OR of ANDs (disjunctive
+                // normal form). That is, rule items will be ANDed together within rule item groups
+                // and the groups themselves will be ORed together.
+                UserListRuleInfo.newBuilder()
+                    .addRuleItemGroups(
+                        UserListRuleItemGroupInfo.newBuilder()
+                            .addAllRuleItems(ImmutableList.of(rule1, rule2))
+                            .build())
+                    .build())
+            .build();
+
+    // Defines a representation of a user list that is generated by a rule.
+    RuleBasedUserListInfo ruleBasedUserListInfo =
+        RuleBasedUserListInfo.newBuilder()
+            // Optional: To include past users in the user list, set the prepopulation_status to
+            // REQUESTED.
+            .setPrepopulationStatus(UserListPrepopulationStatus.REQUESTED)
+            .setExpressionRuleUserList(expressionRuleUserListInfo)
+            .build();
+
+    // Creates the user list.
+    UserList userList =
+        UserList.newBuilder()
+            .setName(StringValue.of("All visitors to example.com" + System.currentTimeMillis()))
```

Comment:
nit: instead of `...example.com" + System.currentTimeMillis()`, do `...example.com #" + System.currentTimeMillis()`, as we do in [AddCampaigns](https://github.com/googleads/google-ads-java/blob/master/google-ads-examples/src/main/java/com/google/ads/googleads/examples/basicoperations/AddCampaigns.java#L153).